### PR TITLE
feat: add product source ADR

### PR DESCRIPTION
**Pull request by AfaqShuaib09**
#### [PROD-3265](https://2u-internal.atlassian.net/browse/PROD-3265)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @AfaqShuaib09!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: decisions/0004-product-source-field.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+Context
+-------
+Owing to the platform focused marketing and revenue strategies, now there is a massive opportunity to improve the marketplace architecture. As muliple LoBs plugged into edx marketplace for marketing their products, the marketplace architecture needs to be improved 
+to support the new business requirements and allow them to create their products through publisher. Currently, only the legacy edx products are created through publisher with empty product source field. 
+In order to maintain the distinction between the different LoBs products, we need to add a new field in the course creation form in publisher to allow the course teams to select the product source for their course.
+
+Decision
+--------
+Adding a new field in the course creation form in publisher to allow the course teams to select the product source for their course. The Product Source field is a drop-down list shown on the course creation page that allows you to select the source of the product.
+As the multiple LoBs plugged to the system, and the products are coming from 3rd party different sources other than legacy edx products, this field is used to identify the source of
```
Comment (Ali-D-Akbar) [inline decisions/0004-product-source-field.rst:18]: * from different 3rd party sources

File: decisions/0004-product-source-field.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+the product. The product source field is hidden for the time being and we are adding product source default value as "edx" for the courses created through publisher on the request of the PCs as Publisher is a self-service tool for all edX partners and it might confuse them
```
Comment (Ali-D-Akbar) [inline decisions/0004-product-source-field.rst:19]: is this gonna be all lower case `edx`? I guess the name of the Source is `edX`.

File: decisions/0004-product-source-field.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+In order to maintain the distinction between the different LoBs products, we need to add a new field in the course creation form in publisher to allow the course teams to select the product source for their course.
+
+Decision
+--------
+Adding a new field in the course creation form in publisher to allow the course teams to select the product source for their course. The Product Source field is a drop-down list shown on the course creation page that allows you to select the source of the product.
+As the multiple LoBs plugged to the system, and the products are coming from 3rd party different sources other than legacy edx products, this field is used to identify the source of
+the product. The product source field is hidden for the time being and we are adding product source default value as "edx" for the courses created through publisher on the request of the PCs as Publisher is a self-service tool for all edX partners and it might confuse them 
+if they see the product source field on the course creation form as the publisher is currently used to publish internal legacy edx products only.
```
Comment (Ali-D-Akbar) [inline decisions/0004-product-source-field.rst:20]: "as the publisher is currently used to publish internal legacy edx products only." seems redundant in the same sentence. Can be removed.

File: decisions/0004-product-source-field.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+Future Enhancements
+-------------------
+For the future enhancements, we are looking to filter the organizations on the basis of the product source and allow the course teams to
+select the organization from the filtered list. Futhermore, different 3rd party products provides different additional metadata fields
```
Comment (Ali-D-Akbar) [inline decisions/0004-product-source-field.rst:25]: *provide

File: decisions/0004-product-source-field.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+for marketing their products. We are looking to add interface where the different partners can ingest their products in bulk using the
```
Comment (Ali-D-Akbar) [inline decisions/0004-product-source-field.rst:26]: *to market their products

Comment (Ali-D-Akbar) [reply decisions/0004-product-source-field.rst:26]: *an interface

Comment (Ali-D-Akbar) [reply decisions/0004-product-source-field.rst:26]: *where different partners can

File: decisions/0004-product-source-field.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+csv upload. On selecting the source of the product, we can give them dummy csv file with the additional metadata fields along with the
```
Comment (Ali-D-Akbar) [inline decisions/0004-product-source-field.rst:27]: *through uploading a CSV

Comment (Ali-D-Akbar) [reply decisions/0004-product-source-field.rst:27]: *by selecting the source of a product

File: decisions/0004-product-source-field.rst [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+--------------------------------------------------------------------
+
+Status
+------
+
+Accepted (March 2023)
+
+Context
+-------
+Owing to the platform focused marketing and revenue strategies, now there is a massive opportunity to improve the marketplace architecture. As muliple LoBs plugged into edx marketplace for marketing their products, the marketplace architecture needs to be improved
```
Comment (Ali-D-Akbar) [inline decisions/0004-product-source-field.rst:11]: owing? 🤔

Comment (Ali-D-Akbar) [reply decisions/0004-product-source-field.rst:11]: *multiple

**Review by Ali-D-Akbar — COMMENTED** [review]
Be sure to run a quick spell/grammar/tenses check before pushing the next version.

File: decisions/0004-product-source-field.rst (hunk 2 above)
Comment (AfaqShuaib09) [reply decisions/0004-product-source-field.rst:19]: yes, product source name is `edX`, though the slug of the product source is `edx`

File: decisions/0004-product-source-field.rst (hunk 7 above)
Comment (AfaqShuaib09) [reply decisions/0004-product-source-field.rst:11]: `Owing` --> `because of` or `due to`

File: decisions/0004-product-source-field.rst [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+In order to maintain the distinction between the different LoBs products, we need to add a new field in the course creation form in publisher to allow the course teams to select the product source for their course.
+
+Decision
+--------
+Adding a new field in the course creation form in publisher to allow the course teams to select the product source for their courses. 
+The Product Source field is a drop-down list that appears on the course creation page, enabling the selection of the source of the product.
+Since multiple LoBs have plugged into the system, and products come from different third-party sources other than legacy edX products, 
+this field is used to identify the source of product. Initially, the product source field will be hidden for the time being, and we will set the default value to "edX" for courses 
+created through the publisher. This decision is based on the request of PCs, as the publisher is a self-service tool for all edX partners, 
+and showing the product source field might confuse them.
```
Comment (DawoudSheraz) [inline decisions/0004-product-source-field.rst:22]: there are some internal-only details here. No need to add them here.

File: decisions/0004-product-source-field.rst [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+to support the new business requirements and allow them to create their products through publisher. Currently, only the legacy edx products are created through publisher with empty product source field. 
+In order to maintain the distinction between the different LoBs products, we need to add a new field in the course creation form in publisher to allow the course teams to select the product source for their course.
+
+Decision
+--------
+Adding a new field in the course creation form in publisher to allow the course teams to select the product source for their courses. 
+The Product Source field is a drop-down list that appears on the course creation page, enabling the selection of the source of the product.
+Since multiple LoBs have plugged into the system, and products come from different third-party sources other than legacy edX products, 
+this field is used to identify the source of product. Initially, the product source field will be hidden for the time being, and we will set the default value to "edX" for courses
```
Comment (DawoudSheraz) [inline decisions/0004-product-source-field.rst:20]: Make sure to emphasize the default value part.

File: decisions/0004-product-source-field.rst [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+this field is used to identify the source of product. Initially, the product source field will be hidden for the time being, and we will set the default value to "edX" for courses 
+created through the publisher. This decision is based on the request of PCs, as the publisher is a self-service tool for all edX partners, 
+and showing the product source field might confuse them.
+
+Future Enhancements
+-------------------
+For future enhancements, we plan to filter organizations based on the product source, allowing course teams to select organizations from a filtered list.
+Additionally, different third-party products provide different additional metadata fields to market their products.
+We are looking to add an interface where different partners can bulk ingest their products through uploading a CSV. By selecting the product source,
+we can provide them with a dummy CSV file with the additional metadata fields along with the required fields for product creation.
```
Comment (DawoudSheraz) [inline decisions/0004-product-source-field.rst:29]: This is unrelated to this at the moment.

**Review by Ali-D-Akbar — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@AfaqShuaib09 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-publisher&entry.867055334=https://github.com/openedx/frontend-app-publisher/pull/858&entry.1484655318=https://github.com/AfaqShuaib09&entry.752974735=2023-03-20+21:05&entry.1917517419=2023-03-27+19:55&entry.2133058324=Yes) so we can improve your experience in the future.
